src/tagged-type.ts

```diff
@@ -0,0 +1,117 @@
+export type Module = string;
+export type DocumentReference = string;
+
+/**
+ * TaggedType provides the encoding/decoding of the Fauna Tagged Type formatting
+ */
+export class TaggedTypeFormat {
+  /**
+   * Encode the Object to the Tagged Type format for Fauna
+   *
+   * @param obj - Object that will be encoded
+   * @returns Map of result
+   */
+  static encode(obj: any): any {
+    return new TaggedTypeEncoded(obj).result;
+  }
+
+  /**
+   * Decode the JSON string result from Fauna to remove Tagged Type formatting.
+   *
+   * @param input - JSON string result from Fauna
+   * @returns object of result of FQL query
+   */
+  static decode(input: string): any {
+    return JSON.parse(input, (_, value: any) => {
+      if (value["@mod"]) {
+        return value["@mod"] as Module;
+      } else if (value["@doc"]) {
+        return value["@doc"] as DocumentReference;
+      } else if (value["@int"]) {
+        return Number(value["@int"]);
+      } else if (value["@long"]) {
+        return Number(value["@long"]);
+      } else if (value["@decimal"]) {
+        return Number(value["@decimal"]);
+      } else if (value["@double"]) {
+        return Number(value["@double"]);
+      } else if (value["@date"]) {
+        return new Date(value["@date"]);
+      } else if (value["@time"]) {
+        return new Date(value["@time"]);
+      } else if (value["@object"]) {
+        return value["@object"];
```

Comment:
I think this should call decode recursively

Reply:
> reviver A function that transforms the results. This function is called for each member of the object.
> If a member contains nested objects, the nested objects are transformed before the parent object is.

shouldn't be necessary